pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -134,14 +134,15 @@ func (gb *GraphBuilder) controllerFor(resource schema.GroupVersionResource, kind
 			gb.graphChanges.Add(event)
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
-			// TODO: check if there are differences in the ownerRefs,
-			// finalizers, and DeletionTimestamp; if not, ignore the update.
 			event := &event{
 				eventType: updateEvent,
 				obj:       newObj,
 				oldObj:    oldObj,
 				gvk:       kind,
 			}
+			if haveSameObjs(event) {
```

Comment:
what is the reason for short-circuiting this?

Reply:
Since the issue has been closed, could help review the PR

Reply:
It's possible for the informer to deliver an updated event without delivering a create event first. In that case, the short circuiting here is not safe. The GC needs to process the update event at this [line](https://github.com/kubernetes/kubernetes/blob/32ec6c212ec9415f604ffc1f4c1f29b782968ff1/pkg/controller/garbagecollector/graph_builder.go#L560).

The performance improvement of this change is unclear. I would just remove the TODO.